website/source/docs/providers/openstack/r/dns_recordset_v2.html.markdown

```diff
@@ -0,0 +1,81 @@
+---
+layout: "openstack"
+page_title: "OpenStack: openstack_dns_recordset_v2"
+sidebar_current: "docs-openstack-resource-dns-recordset-v2"
+description: |-
+  Manages a DNS record set in the OpenStack DNS Service
+---
+
+# openstack\_dns\_recordset_v2
+
+Manages a DNS record set in the OpenStack DNS Service.
+
+## Example Usage
+
+### Automatically detect the correct network
+
+```hcl
+resource "openstack_dns_zone_v2" "example_zone" {
+  name = "example.com."
+  email = "email2@example.com"
+  description = "a zone"
+  ttl = 6000
+  type = "PRIMARY"
+}
+
+resource "openstack_dns_recordset_v2" "rs.example.com" {
+  zone_id = "${openstack_dns_zone_v2.example_zone.id}"
+  name = "example.com."
+  description = "An example record set"
+  ttl = 3000
+  type = "A"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Required) The region in which to obtain the V2 DNS client.
+    If omitted, the `OS_REGION_NAME` environment variable is used.
+    Changing this creates a new DNS  record set.
+
+* `zone_id` - (Required) The ID of the zone in which to create the record set.
+  Changing this creates a new DNS  record set.
+
+* `name` - (Required) The name of the record set. Note the `.` at the end of the name.
+  Changing this creates a new DNS  record set.
+
+* `type` - (Optional) The type of record set. Examples: "A", "MX".
+  Changing this creates a new DNS  record set.
+
+* `ttl` - (Optional) The time to live (TTL) of the record set.
+
+* `description` - (Optional) A description of the  record set.
+
+* `records` - (Optional) An array of DNS records.
+
+* `value_specs` - (Optional) Map of additional options. Changing this creates a
+  new record set.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `region` - See Argument Reference above.
+* `name` - See Argument Reference above.
+* `type` - See Argument Reference above.
+* `ttl` - See Argument Reference above.
+* `description` - See Argument Reference above.
+* `records` - See Argument Reference above.
+* `zone_id` - See Argument Reference above.
+* `value_specs` - See Argument Reference above.
+
+## Import
+
+This resource can be imported by specifying all three arguments, separated
+by a forward slash:
+
+```
+$ terraform import openstack_dns_recordset_v2.recordset_1 <recordset_id>
```

Comment:
So the idea of `terraform import` is to create a `terraform.tfstate` file that is populated by resources that do not exist in a `.tf` file. While `import` will not create the HCL, it will allow Terraform to recognize that those resources exist (so you can base other resources off of them and such).

When importing, you have to specify enough information to bootstrap Terraform to be able to look up the rest of the information. In this case, a Zone ID and RR ID is required. When doing this on the command line as it is now, the following error happens:

```shell
# terraform import openstack_dns_recordset_v2.foo 5805f911-1e86-44b1-8637-dfa73c2d3176
openstack_dns_recordset_v2.foo: Importing from ID "5805f911-1e86-44b1-8637-dfa73c2d3176"...
openstack_dns_recordset_v2.foo: Import complete!
  Imported openstack_dns_recordset_v2 (ID: 5805f911-1e86-44b1-8637-dfa73c2d3176)
openstack_dns_recordset_v2.foo: Refreshing state... (ID: 5805f911-1e86-44b1-8637-dfa73c2d3176)
Error importing: 1 error(s) occurred:

* openstack_dns_recordset_v2.foo (import id: 5805f911-1e86-44b1-8637-dfa73c2d3176): 1 error(s) occurred:

* import openstack_dns_recordset_v2.foo result: 5805f911-1e86-44b1-8637-dfa73c2d3176: import openstack_dns_recordset_v2.foo (id: 5805f911-1e86-44b1-8637-dfa73c2d3176): Terraform detected a resource with this ID doesn't
exist. Please verify the ID is correct. You cannot import non-existent
resources using Terraform import.
```

So we need to use both Zone ID and RR ID. The catch is that the command-line only accepts one argument. So that's where the "separated by a forward slash" comes in.

See the `resource_openstack_compute_floatingip_associate_v2.go` resource for how to handle this. That resource is an edge case because _all_ resource attributes are shoved in the faux ID. Here we'll just do the zone and rr.

Out of all the changes requested, this is the most important as it involves what the ID of the resource will be. It won't be fun to change that later.

If you're wondering why the `_importBasic` acc test passed, it's because all bootstrap information was available in the `_basic` test and it only checks that all resource attributes are populated correctly in the `Read`.

Let me know if you need a hand with this.